src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -275,7 +276,10 @@ protected ConstantValue MakeDefaultExpression(DiagnosticBag diagnostics, Binder
                         valueBeforeConversion, diagnostics, isDefaultParameter: true);
                 }
             }
-
+            if (this.IsNullChecked && convertedExpression.ConstantValue?.IsNull == true)
+            {
+                diagnostics.Add(ErrorCode.WRN_NullCheckedHasDefaultNull, parameterSyntax.Default.Value.Location, this);
```

Comment:
>parameterSyntax.Default.Value.Location [](start = 73, length = 38)

This location looks incorrect. But that should be corrected if this warning is reported in `AddNullCheckingErrorsToParameter` instead.

Reply:
Moving this into the ParameterHelpers helper function causes an infinite loop. It needs to be below the convertedExpression